mucommander-core/src/main/java/com/mucommander/bonjour/BonjourMenu.java

```diff
@@ -90,8 +90,9 @@ public void menuSelected(MenuEvent menuEvent) {
                 // Inform that no service have been discovered
                 add(new JMenuItem(Translator.get("bonjour.no_service_discovered"))).setEnabled(false);
             }
-        }
-        else {
+        } else if (BonjourDirectory.isStarting()) {
+            add(new JMenuItem(Translator.get("bonjour.bonjour_initializing"))).setEnabled(false);
```

Comment:
I think that I never saw an entry in popup menu with indication of something that is being initializing/loading before, I guess it's not common because those menus are not refreshed when the data is available.. I think it would actually be better to wait here until initialization is completed - it would almost never happen in practice.. (that the menu is shown while Bonjour is being initialized)

Reply:
I see it when I open muC and navigate to Bonjour - it initializes in the background for a couple of seconds. And as I stated in PR, for slow/strange networks it make take even longer.

Reply:
that's ok not to block on it during startup but I'd say that when opening the popup menu, it's ok to wait for it to complete rather than indicating that it's being initialized

Reply:
I would rather want to see that Bonjour is still fighting to gather services instead either having whole Bonjour item not present, or the whole popup menu blocked.

Reply:
ok, not that critical - it (opening the popup menu while Bonjour is initializing) should rarely happen